source/features/render-semantic-types.tsx

```diff
@@ -0,0 +1,191 @@
+import React from 'react';
+import {$$} from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+
+import features from '../feature-manager.js';
+
+function isSemanticCommitTitleElement(commitTitle: string): boolean {
+	// This may not be fully accurate.
+	// Tests, if the title string
+	// starts with a word,
+	// followed by an anything in parentheses,
+	// followed by an optional exclamation mark,
+	// followed by a colon,
+	// followed by anything.
+	return /^\w*(?:\(.*\))?!?:.*/.test(commitTitle);
```

Comment:
We need to extract this regex and add tests. You can create:

- source/helpers/semantic-release.ts
- source/helpers/semantic-release.test.ts

I’d want to use a single regex for parsing and extraction of the data to make sure that these two parts are never out of sync. You should use regex named groups for this.

Reply:
We should include this link also, for documentation: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#-commit-message-format

Reply:
> I’d want to use a single regex for parsing and extraction of the data to make sure that these two parts are never out of sync. You should use regex named groups for this.

Sometimes I only want to parse the scope, so I can not reuse the regex that captures the type+ optional scope.

Reply:
Why you can’t? Regex can do it

Reply:
I'll try to resolve it.

Reply:
Something like this, the scope is also optional:

![IMG_5350](https://github.com/user-attachments/assets/a1f2af14-840d-4590-afcd-01b3b97d67d9)